bin/prepare_feedback.py

```diff
@@ -0,0 +1,173 @@
+            f.write("\n\n")
+
+
+def fix_tutorial_info(df):
+    '''Change tutorial topic or title
+
+    :param df: dataframe    
+    '''
+    df = df.copy()
+    # change topic for some tutorials
+    for tuto in new_topic_for_tuto:
+        df.loc[df.tutorial == tuto, 'topic'] = new_topic_for_tuto[tuto]
+    # rename topic for all tutorials in a topic
+    for topic in new_topics:
+        df.topic = (df
+            .topic
+            .replace(to_replace=topic, value=new_topics[topic]))
+    # rename some tutorials
+    for tuto in new_tutorial_name:
+        df.loc[df.tutorial == tuto, 'tutorial'] = new_tutorial_name[tuto]
+    return df
+
+
+def extract_topic_tutorial_name(df):
+    '''Extract topic from tutorial name
+
+    :param df: dataframe
+    '''
+    df = df.copy()
+    new = df['tutorial_topic'].str[::-1].str.split('(', n = 1, expand = True)
+    df["tutorial"]= new[1].str[::-1].str[:-1]
+    df["topic"]= new[0].str[::-1].str[:-1]
+    return df
+
+
+def prepare_feedback(url, out_file):
+    '''Get and prepare feedback CSV file
+
+    :param url: URL to Google sheet with feedback answers
+    :param out_file: Path to output file
+    '''
+    df = (pd.read_csv(url, sep='\t')
+        # rename column
+        .rename(columns = {'Timestamp': 'timestamp',
+                            'How much did you like this tutorial?': 'note',
+                            'What did you like?': 'pro',
+                            'What could be improved?': 'con',
+                            'Tutorial': 'tutorial_topic',
+                            'Your feedback is always anonymous. Also make it confidential (only visible to admins)?': 'anonymous'})
+        # extract topic from tutorial name
+        .pipe(extract_topic_tutorial_name)
+        # remove rows with NaN on note, pro and con
+        .dropna(subset=['note', 'pro', 'con'], how='all')
+        # replace NaN in note by 0
+        .fillna(value={'note': 0})
+        # fill other NaN by empty string
+        .fillna('')
+        # format 
+        .assign(
+            #note to integer
+            note=lambda x: x['note'].astype(int),
+            # format pro and con to string
+            pro=lambda x: x['pro'].astype(str),
+            con=lambda x: x['con'].astype(str),
+            # extract month and date
+            timestamp=lambda x: pd.to_datetime(x['timestamp'], dayfirst=True))
+        .assign(
+            month=lambda x: x['timestamp'].dt.strftime("%Y-%m"),
+            date=lambda x: x['timestamp'].dt.strftime("%Y-%m-%d"))
+        # change topic for some tutorials
+        .pipe(fix_tutorial_info)
+        # remove extra columns
+        .drop(columns=["tutorial_topic", "timestamp"])
+        # remove some rows 
+        .query('topic != "TESTIN"')
+        .query('topic != "tes"')
+        .query('topic != ""')
+        # remove tutorials
+        .query('tutorial != "RNA-seq counts to genes and pathways"')
+        .query('tutorial != "Galaxy Tool Management"')
```

Comment:
any reason why are these being removed? was the data bad or something? it isn't clear for me.

Reply:
I could not find these tutorials anymore. I think they were renamed or completely reshaped into new ones

Reply:
Galaxy tool management is this one: https://training.galaxyproject.org/training-material/topics/admin/tutorials/tool-management/tutorial.html, job metrics was deprecated, yeah.